### PR TITLE
WD-1180 - Create Juju slice

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

Create Juju slice.
Add missing tests.

## Details

https://warthogs.atlassian.net/browse/WD-1180

### CODE REVIEW

File: src/store/juju/actions.test.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
+import { actions } from "./slice";
+
+const status = {
```
Comment (huwshimi) [inline src/store/juju/actions.test.ts:3]: These objects will be replaced with factories.

File: src/store/juju/actions.test.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    ).toStrictEqual({
+      type: "juju/updateModelList",
+      payload: {
+        models,
+        wsControllerURL: "wss://test.example.com",
+      },
+    });
+  });
+
+  it("updateModelStatus", () => {
```
Comment (huwshimi) [inline src/store/juju/actions.test.ts:63]: From here down are new tests.

File: src/store/juju/reducers.test.ts [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+  annotations: undefined,
+  applications: status.applications,
+  machines: status.machines,
+  model: status.model,
+  offers: status.offers,
+  relations: status.relations,
+  "remote-applications": status["remote-applications"],
+};
+
+describe("reducers", () => {
```
Comment (huwshimi) [inline src/store/juju/reducers.test.ts:55]: These tests are all new.

File: src/store/juju/selectors.test.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+} from "./selectors";
+
+const defaultState = {
+  controllers: null,
+  models: {},
+  modelData: {},
+  modelWatcherData: {},
+};
+
+describe("selectors", () => {
```
Comment (huwshimi) [inline src/store/juju/selectors.test.ts:23]: These tests are all new.

File: src/store/juju/thunks.test.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import { rootStateFactory } from "testing/factories";
+
+import { addControllerCloudRegion } from "./thunks";
+import { actions } from "./slice";
+
+describe("thunks", () => {
```
Comment (huwshimi) [inline src/store/juju/thunks.test.ts:6]: This test is new.

**Review by goulinkh — APPROVED** [review]

**Pull request merged**